### PR TITLE
Feature/zabbix agent

### PR DESCRIPTION
**Pull request by Talits**

### CODE REVIEW

**Comment by Talits** [comment]
[[Installer | UOL] Incluir Zabbix Agent no Ansible para monitorar infra](https://trello.com/c/yAy0I1by/96-installer-uol-incluir-zabbix-agent-no-ansible-para-monitorar-infra)

File: ansible/playbooks/getup-monitoring/config.yml [hunk 1]
```diff
@@ -47,3 +47,14 @@
...
+
+- name: Zabbix-Agent
+  hosts: oo_first_master
+  gather_facts: no
+  tags:
+    - getup-zabbix
+  tasks:
+    - include_role:
+        name: getup-zabbix-agent
+      when: getupcloud_install_alertmanager_zabbix_webhook | default(false) | bool
```
Comment (caruccio) [inline ansible/playbooks/getup-monitoring/config.yml]: It should depends on a differente var than zabbix alertmanager webhook. Those are not the same.

File: ansible/roles/defaults/main.yml [hunk 2]
```diff
@@ -109,6 +109,7 @@ getupcloud_zabbix_host_annotation: "zabbix_host"
 getupcloud_alertmanager_zabbix_webhook_replicas: 2
 zabbix_url: zabbix_server_url
 
+
```
Comment (caruccio) [inline ansible/roles/defaults/main.yml]: Is it really necessary?

File: ansible/roles/getup-zabbix-agent/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+- name: Docker Zabbix | Create DaemonSet
+  oc_process:
+    namespace: "{{ openshift_prometheus_namespace }}"
+    template_name: dockbix-agent-zabbix
+    create: True
+
+- name: Permission | Set Permissions SCC for dockbix-agent-zabbix
+  oc_adm_policy_user:
+    state: present
+    user: "system:serviceaccount:{{ openshift_prometheus_namespace }}:dockbix-agent-zabbix"
```
Comment (caruccio) [inline ansible/roles/getup-zabbix-agent/tasks/main.yml]: Where is this `sa` being created?

File: ansible/roles/defaults/main.yml (hunk 2 above)
Comment (Talits) [reply ansible/roles/defaults/main.yml]: yes , the agent use the server endpoint

File: ansible/roles/getup-zabbix-agent/tasks/main.yml (hunk 3 above)
Comment (Talits) [reply ansible/roles/getup-zabbix-agent/tasks/main.yml]: in the namespace prometheus - Template create sa

File: ansible/playbooks/getup-monitoring/config.yml (hunk 1 above)
Comment (Talits) [reply ansible/playbooks/getup-monitoring/config.yml]: ok

File: ansible/roles/defaults/main.yml (hunk 2 above)
Comment (guilhermesteves) [reply ansible/roles/defaults/main.yml]: Me, in my innocence, reviewing things in portuguese

File: ansible/roles/getup-zabbix-agent/templates/zabbix-docker.yaml.j2 [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+              requests:
+                cpu: 0.15
+            securityContext:
+              privileged: true
+            env:
+              - name: ZA_Server
+                value: {{ zabbix_url }}
+              - name: ZA_ServerActive
+                value: {{ zabbix_url }}
+            image: monitoringartist/zabbix-agent-xxl-limited
```
Comment (caruccio) [inline ansible/roles/getup-zabbix-agent/templates/zabbix-docker.yaml.j2]: Let's try official image from https://hub.docker.com/r/zabbix/zabbix-agent/

Comment (Talits) [reply ansible/roles/getup-zabbix-agent/templates/zabbix-docker.yaml.j2]: ok

File: ansible/roles/defaults/main.yml (hunk 2 above)
Comment (Talits) [reply ansible/roles/defaults/main.yml]: :)

**Comment by Talits** [comment]
ok

File: ansible/roles/getup-zabbix-agent/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
+---
+# tasks file for docker-zabbix-agent
+- name: Docker Zabbix | Create Service Account
+  oc_serviceaccount:
+    name: agent-zabbix
```
Comment (caruccio) [inline ansible/roles/getup-zabbix-agent/tasks/main.yml]: Better have a consistent naming. Suggest change it to `zabbix-agent` (what-who)

File: ansible/roles/getup-zabbix-agent/templates/zabbix-docker.yaml.j2 [hunk 6]
```diff
@@ -0,0 +1,43 @@
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: agent-zabbix
+  labels:
+    tier: monitoring
+    app: agent-zabbix
```
Comment (caruccio) [inline ansible/roles/getup-zabbix-agent/templates/zabbix-docker.yaml.j2]: Also keep-consistency.

File: ansible/roles/getup-zabbix-agent/templates/zabbix-docker.yaml.j2 [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    version: v1
+spec:
+  template:
+    metadata:
+      labels:
+        name: agent-zabbix
```
Comment (caruccio) [inline ansible/roles/getup-zabbix-agent/templates/zabbix-docker.yaml.j2]: Also keep-consistency.

File: ansible/roles/getup-zabbix-agent/templates/zabbix-docker.yaml.j2 [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    spec:
+      hostPID: true
+      hostIPC: true
+      hostNetwork: false
+      serviceAccountName: agent-zabbix
```
Comment (caruccio) [inline ansible/roles/getup-zabbix-agent/templates/zabbix-docker.yaml.j2]: You got the idea ;)

**Comment by Talits** [comment]
done

**Pull request merged**